1.cellprofiler_ic/image_quality_control/scripts/1.evaluate_qc.py

```diff
@@ -0,0 +1,391 @@
+    )
+    plt.close()  # Close the plot to prevent it from being displayed
+
+
+# ### Generate density plot with all plates together and save
+
+# In[8]:
+
+
+# Number of plates
+num_plates = len(df["Metadata_Plate"].unique())
+
+# Calculate the number of rows and columns for the subplots
+num_rows = 2  # You want 2 rows for the quad order
+num_cols = (num_plates + 1) // num_rows  # +1 to round up
+
+# Set the style
+sns.set_style("whitegrid")
+
+# Create subplots with custom layout
+fig, axes = plt.subplots(nrows=num_rows, ncols=num_cols, figsize=(15, 5 * num_rows))
+
+# Flatten the axes array for easier indexing
+axes = axes.flatten()
+
+# Iterate through each plate
+for idx, plate in enumerate(df["Metadata_Plate"].unique()):
+    # Plot all channels in the same subplot
+    ax = axes[idx]
+    plate_df = df[df["Metadata_Plate"] == plate]
+    sns.kdeplot(
+        data=plate_df,
+        x="ImageQuality_PowerLogLogSlope",
+        hue="Channel",
+        palette=["b", "g", "r", "magenta"],
+        fill=True,
+        common_norm=False,
+        ax=ax,
+    )
+
+    # Set vertical lines at -2.5 and -1.25
+    ax.axvline(x=threshold_value_above_mean, color="red", linestyle="--")
+    ax.axvline(x=threshold_value_below_mean, color="red", linestyle="--")
+
+    ax.set_title(f"Density plots for {plate}")
+    ax.set_xlabel("ImageQuality_PowerLogLogSlope")
+    ax.set_ylabel("Density")
+
+    # Set the x-axis range for the current subplot
+    ax.set_xlim(-4.0, 0)
+
+# Remove empty subplots
+for i in range(num_plates, len(axes)):
+    fig.delaxes(axes[i])
+
+# Adjust layout
+plt.tight_layout()
+
+# Save the plot
+plt.savefig(figure_dir / "all_channels_combined_density.png", dpi=500)
+
+plt.close()  # Close the plot to prevent it from being displayed
+
+
+# ### Visualize the distribution of the identified outliers
+
+# In[9]:
+
+
+# Plot outliers as a distribution plot with hue as 'Metadata_Plate'
+sns.set_style("whitegrid")
+plt.figure(figsize=(12, 8))
+
+# Using kdeplot (you can also use distplot)
+sns.kdeplot(
+    data=blur_outliers,
+    x="ImageQuality_PowerLogLogSlope",
+    hue="Metadata_Plate",
+    palette="viridis", 
+    fill=True,
+    common_norm=False,
+)
+
+# Set vertical lines at -2.5 and 1.25
+plt.axvline(x=threshold_value_above_mean, color="red", linestyle="--")
+plt.axvline(x=threshold_value_below_mean, color="red", linestyle="--")
+
+plt.title("Distribution of Outliers per Plate")
+plt.xlabel("ImageQuality_PowerLogLogSlope")
+
+plt.tight_layout()
+
+plt.savefig(figure_dir / "blur_outliers_per_plate.png", dpi=500)
+
+plt.show()
+
+
+# ## Saturation metric
+# 
+# For saturation metrics, we are looking for:
+# 
+# - Smudged images or images containing large artifacts (usually seen in `DNA` channel regardless of stain)
+# - Overly saturated channels which can occur in any channel, but specifically we are seeing `Mito` having the most problems with this.
```

Comment:
Are you saying Mito has the most saturated images?

Reply:
Correct! I have noticed that the Mito channel suffers most from over-saturation, which I do believe is due to some staining issues or microscope parameter issues.